rcl_lifecycle/CMakeLists.txt

```diff
@@ -11,6 +11,21 @@ find_package(rmw REQUIRED)
 
 include_directories(include)
 
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(NOT WIN32)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wpedantic")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
```

Comment:
These two lines could become a one-liner by using `add_compile_options`.
Also I think the previous condition was more accurate: `CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"` because these flags are compiler dependant rather than Platform dependant